packages/analytics-js-plugins/src/deviceModeTransformation/index.ts

```diff
@@ -17,25 +39,159 @@ const DeviceModeTransformation = (): ExtensionPlugin => ({
+      const ACTION_TO_DROP_EVENT = 'Dropping the event';
+      const destinations = state.nativeDestinations.dmtEnabledDestinations.value;
+
+      destinations.forEach(dest => {
+        try {
+          const eventsToSend: TransformedEvent[] = [];
+          switch (status) {
+            case 200: {
+              const response = JSON.parse(result);
+              const destTransformedResult = response.transformedBatch.find(
+                (e: TransformedBatch) => e.id === dest.id,
+              );
+              destTransformedResult?.payload.forEach((tEvent: TransformedPayload) => {
+                if (tEvent.status === '200') {
+                  eventsToSend.push(tEvent.event);
+                } else {
+                  let reason = 'Unknown';
+                  if (tEvent.status === '410') {
+                    reason = 'Transformation is not available';
+                  }
+
+                  let action = ACTION_TO_DROP_EVENT;
+                  if (dest.propagateEventsUntransformedOnError === true) {
+                    action = ACTION_TO_SEND_UNTRANSFORMED_EVENT;
+                    eventsToSend.push(event);
+                    logger?.warn(
+                      DMT_TRANSFORMATION_UNSUCCESSFUL_ERROR(dest.displayName, reason, action),
+                    );
+                  } else {
+                    logger?.error(
+                      DMT_TRANSFORMATION_UNSUCCESSFUL_ERROR(dest.displayName, reason, action),
+                    );
+                  }
+                }
+              });
+
+              break;
+            }
+            // Transformation server access denied
+            case 404: {
+              logger?.warn(DMT_SERVER_ACCESS_DENIED_WARNING());
+              eventsToSend.push(event);
+              break;
+            }
+            default: {
+              if (dest.propagateEventsUntransformedOnError === true) {
+                logger?.warn(
+                  DMT_REQUEST_FAILED_ERROR(
+                    dest.displayName,
+                    status,
+                    ACTION_TO_SEND_UNTRANSFORMED_EVENT,
+                  ),
+                );
+                eventsToSend.push(event);
+              } else {
+                logger?.error(
+                  DMT_REQUEST_FAILED_ERROR(dest.displayName, status, ACTION_TO_DROP_EVENT),
+                );
+              }
+              break;
+            }
+          }
+          eventsToSend?.forEach((tEvent?: TransformedEvent) => {
+            if (isNonEmptyObject(tEvent)) {
+              pluginsManager.invokeSingle(
+                NATIVE_DEST_EXT_POINT,
+                state,
+                tEvent,
+                dest,
+                errorHandler,
+                logger,
+              );
+            }
+          });
+        } catch (e) {
+          if (e instanceof Error) {
+            e.message = DMT_EXCEPTION(dest.displayName, e.message);
```

Comment:
There is no need to alter the error object. You can see the error object and a custom message to the error handler.

Reply:
Done